src/main/main.ts

```diff
@@ -917,6 +917,24 @@ const createWindow = async () => {
         mainWindow.webContents.openDevTools();
     }
 
+    let hasUnsavedChanges = false;
+    ipcMain.on('update-has-unsaved-changes', (_, value) => {
+        hasUnsavedChanges = value;
+    });
+
+    mainWindow.on('close', (event) => {
+        if (hasUnsavedChanges) {
+            const choice = dialog.showMessageBoxSync(mainWindow, {
+                type: 'question',
+                buttons: ['Yes', 'No'],
+                title: 'Discard unsaved changes?',
+                message:
+                    'The current chain has some unsaved changes. Do you really want to quit without saving?',
+            });
+            if (choice > 0) event.preventDefault();
```

Comment:
`choice` is just an index of the `buttons` array, so please be explicit.
```suggestion
            if (choice === 1) event.preventDefault();
```

Reply:
That's what I get for copying and pasting from SO lol